File: 118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> array(numRows);
        for (int i = 0; i < numRows; i++){
            array[i].resize(i+1);
            for (int j = 0; j <= i; j ++){
                if (j == 0){
                    array[i][j] = 1;
                }
                else if (j == i){
                    array[i][j] = 1;
                }
                else{
                    array[i][j] = array[i-1][j-1] + array[i-1][j];
                }
            }
        }
        return array;
    }
};

File: 118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        arr = [[0 for j in range(1, i + 1)] for i in range(1, numRows + 1)]
        for i in range(numRows):
            arr[i][0] = 1
            arr[i][-1] = 1
            if i >= 2:
                for j in range(1, len(arr[i]) - 1):
                    arr[i][j] = arr[i-1][j-1] + arr[i-1][j]
        return arr